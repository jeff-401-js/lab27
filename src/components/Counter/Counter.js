import React from "react";


class Counter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: 0
    };
  }

  handleButtonClick = e => {
    e.preventDefault();
    this.setState({ counter: this.state.counter + 1 });
  };

  handleButtonClickDown = e => {
    e.preventDefault();
    this.setState({ counter: this.state.counter - 1 });
  };

  render() {
    let counter = this.state.counter;
    let color;
    let colorRed = { color: "red" };
    let colorBlue = { color: "blue" };
    if (counter > 0) {
      color = colorBlue;
    } else if (counter < 0) {
      color = colorRed;
    }

    return (
      <div>
        <h4 style={color}>{this.state.counter}</h4>
        <br />
        <a href="*" onClick={this.handleButtonClick}>
          +Click+Me+
        </a>
        <br />
        <br />
        <a href="*" onClick={this.handleButtonClickDown}>
          -Click-Me-
        </a>
      </div>
    );
  }
}

export default Counter;