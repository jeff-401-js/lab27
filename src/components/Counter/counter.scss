@import '../../app.scss';

div{
  div :hover{
    background-color: rgb(28, 61, 42);
  }
  h4 {
    padding-left: 10px;
    max-width: 50px;
    font-size: 50px;
    color: rgb(0, 255, 213);
    font-family: $mainFont;
    background-color: $backGroundColor;
    box-shadow: 2px 5px black;
  }
  a {
    max-width: 120px;
    font-size: 30px;
    color: rgb(224, 19, 146);
    font-family: $mainFont;
    background-color: $backGroundColor;
  }
}