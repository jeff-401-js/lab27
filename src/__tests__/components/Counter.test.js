/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../../components/Counter/Counter';


describe('<Counter /> Enzyme Test', () => {
  it('is alive at application start', () => {
    // eslint-disable-next-line no-undef
    const app = shallow(<Counter/>);
    expect(app.find('.count').text()).toBe('0');
  });

  it('can count up', () => {
    const wrapper = mount(<Counter/>);

    wrapper.find('.up').simulate('click');
    expect(wrapper.state('counter')).toBe(1);
    wrapper.find('.up').simulate('click');
    expect(wrapper.state('counter')).toBe(2);
  });

  it('can count down', () => {
    const wrapper = mount(<Counter/>);

    wrapper.find('.down').simulate('click');
    expect(wrapper.state('counter')).toBe(-1);
    wrapper.find('.down').simulate('click');
    expect(wrapper.state('counter')).toBe(-2);
  });
});

describe('<Counter/> Snapshot Test', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Counter/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
