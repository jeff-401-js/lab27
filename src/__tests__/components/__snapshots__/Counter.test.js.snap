// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> Snapshot Test renders correctly 1`] = `
<div
  className="counter"
>
  <h4
    className="count"
  >
    0
  </h4>
  <br />
  <a
    className="up"
    href="*"
    onClick={[Function]}
  >
    +Click+Me+
  </a>
  <br />
  <br />
  <a
    className="down"
    href="*"
    onClick={[Function]}
  >
    -Click-Me-
  </a>
</div>
`;
